src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -110,44 +121,126 @@ protected override void Dispose(bool disposing)
 
         #region Private methods
 
-        private void StartRequest(RequestCompletionSource state)
+        private async void StartRequest(RequestCompletionSource state)
```

Comment:
> The curl_easy_perform API does not return completion till headers AND body are received

At a higher-level, then, why are we using the "easy" functions?  I just looked at the libcurl docs, which state:

```
The easy interface is a synchronous, efficient, quickly used and... yes, easy interface for file transfers. Numerous applications have been built using this.

The multi interface is the asynchronous brother in the family and it also offers multiple transfers using a single thread and more. Get a grip of how to work with it in the multi interface overview.
```

So shouldn't we be using the multi interface rather than the easy interface?  I'm concerned about the current path because it's going to block a thread for the duration of the HTTP call.  That's very non-scalable, and scalability will be critical in a server environment.  The docs for the multi interface state in its objectives ```Enable event-based handling and scaling transfers up to and beyond thousands of parallel connections.```, which is a key need of this library as well.